src/corelibs/U2View/src/ov_msa/general_tab/MaConsensusModeWidget.cpp

```diff
@@ -44,6 +44,26 @@ MaConsensusModeWidget::MaConsensusModeWidget(QWidget* parent)
     setupUi(this);
 }
 
+void MaConsensusModeWidget::reInit(MultipleAlignmentObject* _maObject, MaEditorConsensusArea* _consArea) {
+    SAFE_POINT(_maObject != nullptr, "MaConsensusModeWidget can not be initialized: MultipleAlignmentObject is NULL", );
+    SAFE_POINT(_consArea != nullptr, "MaConsensusModeWidget can not be initialized: MaEditorConsensusArea is NULL", );
+
+    consArea = _consArea;
+    maObject = _maObject;
+
+    initConsensusTypeCombo();
+
+    connect(this, SIGNAL(si_algorithmChanged(QString)), consArea, SLOT(sl_changeConsensusAlgorithm(QString)));
+    connect(this, SIGNAL(si_thresholdChanged(int)), consArea, SLOT(sl_changeConsensusThreshold(int)));
+
+    connect(consArea,
```

Comment:
For any new signals connector use compile time checked function pointers but not runtime-checked SIGNAL/SLOT macros

Reply:
done for this